hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -0,0 +1,126 @@
+    private final GlobalDynamicProperties dynamicProperties;
+
+    /**
+     * Constructor of the {@code IngestChecker}
+     *
+     * @param transitionLogic a {@link TransitionLogicLookup}
+     * @param dynamicProperties the {@link GlobalDynamicProperties}
+     * @throws NullPointerException if one of the arguments is {@code null}
+     */
+    public IngestChecker(
+            @NonNull final TransitionLogicLookup transitionLogic,
+            @NonNull final GlobalDynamicProperties dynamicProperties) {
+        this.transitionLogic = requireNonNull(transitionLogic);
+        this.dynamicProperties = requireNonNull(dynamicProperties);
+    }
+
+    /**
+     * Checks a transaction for semantic errors
+     *
+     * @param txBody the {@link TransactionBody}
+     * @param functionality the {@link HederaFunctionality} of the transaction
+     * @throws NullPointerException if one of the arguments is {@code null}
+     * @throws PreCheckException if a semantic error was discovered. The contained {@code
+     *     responseCode} provides the error reason.
+     */
+    public void checkTransactionSemantic(
+            @NonNull final TransactionBody txBody, @NonNull final HederaFunctionality functionality)
+            throws PreCheckException {
+        final ResponseCodeEnum errorCode;
+        if (functionality == TokenAccountWipe) {
+            // TODO: Not really sure why TokenAccountWipe gets special treatment
+            errorCode =
+                    TokenWipeAccessor.validateSyntax(
+                            txBody.getTokenWipe(),
+                            dynamicProperties.areNftsEnabled(),
+                            dynamicProperties.maxBatchSizeWipe());
+        } else {
+            final var logic =
+                    transitionLogic
+                            .lookupFor(functionality, txBody)
+                            .orElseThrow(() -> new PreCheckException(NOT_SUPPORTED));
+            errorCode = logic.semanticCheck().apply(txBody);
+        }
+        if (errorCode != OK) {
+            throw new PreCheckException(errorCode);
+        }
+    }
+
+    /**
+     * Checks the signature of the payer. <em>Currently not implemented.</em>
+     *
+     * @param txBody the {@link TransactionBody}
+     * @param signatureMap the {@link SignatureMap} contained in the transaction
+     * @param payer the {@code Account} of the payer
+     * @throws NullPointerException if one of the arguments is {@code null}
+     * @throws PreCheckException if an error is found while checking the signature. The contained
+     *     {@code responseCode} provides the error reason.
+     */
+    public void checkPayerSignature(
+            @NonNull final TransactionBody txBody,
+            @NonNull final SignatureMap signatureMap,
+            @NonNull final Object /* Account */ payer)
+            throws PreCheckException {
+        LOG.warn("IngestChecker.checkPayerSignature() has not been implemented yet");
```

Comment:
Can throw `NotImplementedException` instead of warn message

Reply:
Same as before, I want these checks to pass even before the functionality was implemented.